src/huggingface_hub/repository.py

```diff
@@ -641,15 +642,20 @@ def clone_from(self, repo_url: str, use_auth_token: Union[bool, str, None] = Non
 
                 scheme = urlparse(repo_url).scheme
                 repo_url = repo_url.replace(f"{scheme}://", f"{scheme}://user:{token}@")
-
                 if namespace == user or namespace in valid_organisations:
-                    self.client.create_repo(
-                        repo_id=repo_id,
-                        token=token,
-                        repo_type=self.repo_type,
-                        exist_ok=True,
-                        private=self.private,
-                    )
+                    try:
+                        _ = HfApi().repo_info(
+                            f"{repo_id}", repo_type=self.repo_type, token=token
+                        )
+                    except HTTPError:
+                        self.client.create_repo(
+                            repo_id=repo_id,
+                            token=token,
+                            repo_type=self.repo_type,
+                            exist_ok=True,
+                            private=self.private,
```

Comment:
Wouldn't this also need a `space_sdk` in case the repository in question is a space? i.e., if the repo is a space that doesn't exist, it'll try to create it but fail at doing so.

As we're deprecating, maybe that's intended and we keep that use-case as it is (i.e., not working)

Reply:
I will raise a `ValueError` in case someone tries to create Space that doesn't exist because otherwise I have to add `space_sdk` as a class attribute to `Repository` (we're later deprecating the repository creation in `clone_from` anyway so there's no point to add that). I'll add two tests, one for model repo and one for Space repo to see if the error is raised.  We can merge tomorrow ^^

Reply:
Sounds great, thank you @merveenoyan!